engine/src/main/java/org/terasology/rendering/dag/RenderTaskListGenerator.java

```diff
@@ -25,29 +28,48 @@
  */
 public final class RenderTaskListGenerator {
     private static final Logger logger = LoggerFactory.getLogger(RenderTaskListGenerator.class);
+    private List<Object> intermediateList;
+    private List<RenderPipelineTask> taskList;
 
     public RenderTaskListGenerator() {
+        taskList = Lists.newArrayList();
+        intermediateList = Lists.newArrayList();
     }
 
     public List<RenderPipelineTask> generateFrom(List<Node> orderedNodes) {
-        logger.info("Task List: "); // TODO: remove in the future or turn it into debug
-        List<Object> intermediateList = generateIntermediateList(orderedNodes);
-        return generateTaskList(intermediateList);
+        generateIntermediateList(orderedNodes);
```

Comment:
I suspect we'll need the intermediate list to be somehow available to other objects, i.e. nodes, to remove their StateChanges when they are removed from the graph. But for the time being we can leave it here, hidden from sight.